jetty-io/src/main/java/org/eclipse/jetty/io/ManagedSelector.java

```diff
@@ -385,6 +384,8 @@ private boolean select()
                     if (LOG.isDebugEnabled())
                         LOG.debug("Selector {} waiting on select", selector);
                     int selected = selector.select();
+                    if (selected == 0)
+                        selected = selector.selectNow();
```

Comment:
This is the key fix for the bug, identified during a conversation with @gregw and @sbordet